src/classes/CON_PreferredPhone_Test.cls

```diff
@@ -0,0 +1,450 @@
+     any other Phone values and when Preferred Phone is Blank and the Context is batch.
+    */
+
+    @isTest 
+    static void testBatchPhoneNull() {
+         
+        UTIL_CustomSettingsFacade.getSettingsForTests(
+            new Hierarchy_Settings__c(Disable_Preferred_Phone_Enforcement__c = True)
+            );
+        
+        Contact newCont = new Contact(
+            FirstName = 'JohnnyTest1',
+            LastName = 'JohnnyTest',
+            Phone  = '111111111',
+            OtherPhone  = '0000000000'
+        );
+        insert newCont;
+        
+        
+        Test.startTest();
+            UTIL_CustomSettingsFacade.getSettingsForTests(
+            new Hierarchy_Settings__c(Disable_Preferred_Phone_Enforcement__c = False)
+            );
+            CON_Phone_BATCH batch = new CON_Phone_BATCH(null);
+             Database.executeBatch(batch);
+        Test.stopTest();
+
+        Contact newContAfterList = [SELECT Id, FirstName, LastName, Name, PreferredPhone__c  FROM Contact];
+        
+        System.assertEquals('Phone (standard)', newContAfterList.PreferredPhone__c);
+
+    }   
+    
+    /* 
+     Updates Preferred Phone to the label of the Phone field that has a value which is not same as 
+     standard phone
+    */
+
+    @isTest 
+    static void testBatchOther() {
+         
+        UTIL_CustomSettingsFacade.getSettingsForTests(
+            new Hierarchy_Settings__c(Disable_Preferred_Phone_Enforcement__c = True)
+            );
+        
+        Contact newCont = new Contact(
+            FirstName = 'JohnnyTest1',
+            LastName = 'JohnnyTest',
+            Phone  = '111111111',
+            OtherPhone  = '111111111'
+        );
+        insert newCont;
+        
+        
+        Test.startTest();
+            UTIL_CustomSettingsFacade.getSettingsForTests(
+            new Hierarchy_Settings__c(Disable_Preferred_Phone_Enforcement__c = False)
+            );
+            CON_Phone_BATCH batch = new CON_Phone_BATCH(null);
+             Database.executeBatch(batch);
+        Test.stopTest();
+
+        Contact newContAfterList = [SELECT Id, FirstName, LastName, Name, PreferredPhone__c  FROM Contact];
+                
+        System.assertEquals('Other', newContAfterList.PreferredPhone__c);
+       
+    }   
+    
+    /*  Validation when there are more than one phone fields but not Preferred Phone */
+    @isTest 
+    static void testErrorPrefBlank() {
+        
+        try {
+            Contact newCont = new Contact(
+                FirstName = 'JohnnyTest1',
+                LastName = 'JohnnyTest',
+                WorkPhone__c  = '111111111',
+                OtherPhone  = '0000000000'
+        );
+            Test.startTest();
+            insert newCont;
+            
+            Test.stopTest();
+            
+        }catch(Exception error) {
+             System.assert(error.getMessage().contains('Tell us which Phone is preferred.'));
```

Comment:
Since this is a label, to make this test dynamic (in case we update the label later), can we do `.contains(Label.theNameOfTheLabelHere)` ?